align/schema/instance.py

```diff
@@ -45,6 +49,16 @@ def mclass(self):
         assert model is not None, self.parent.parent.parent
         return model
 
+    @types.validator('name', allow_reuse=True)
+    def unique_name(cls, name):
+        name = name.upper()
+        # Need help in fixing this code
+        # assert isinstance(cls._validator_ctx().parent, List[Instance]), 'Instance can only be instanitated within List[Instance]'
+        # if cls._validator_ctx().parent.__root__:
+        #     all_names = {inst.name for inst in cls._validator_ctx().parent}
+        #     assert name not in all_names
```

Comment:
This doesn't look fully implemented  yet.

Reply:
Yes. I wanted to check for repeat names in the list but I am getting an error 
AttributeError: 'List[Instance]' object has no attribute '__root__'
Any idea how to resolve this?

Reply:
Which line? I don't know if line 56 will work as expected but it doesn't look like its mandatory for the actual check.

Reply:
all_names = {inst.name for inst in cls._validator_ctx().parent}

Reply:
Okay. So I suspect that this check cannot be implemented here. It looks like this check is being called before the parent List[Instance] object has been fully instantiated.

There are two ways to fix this:
1) Create an empty list and append to it from the parent constructor (subcircuit in this case)
2) Move the check to where it belongs (subcircuit) as that's where you care about non-repetition.

I would go with option 2 as non duplication of 'name' is not something that any given Instance cares about anyway. Its the subcircuit that cares if the instance name is duplicated. The name.upper() casting can stay here.

Reply:
The check at the subcircuit level would look a little different of course. Something like `len(value) == len({v.name for v in value})`

Reply:
done